### ISSUE TITLE
Add formatting support for informational items in contextual links

### ISSUE THREAD

**Issue by bugfolder** [backdrop/backdrop-issues#5274]
**Description of the need**

There is a use case for putting informational items into a list of contextual links: for example, putting the title of a Layout (or Block, View, etc.) at the top of the contextual links list. See [this comment](https://github.com/backdrop/backdrop-issues/issues/2626#issuecomment-929384914) for an example.

This can be done by just adding the text of the information as an element of the links array. However, the formatting of that list in Contextual module expects an `<a>` tag on the element, so if this is done with ordinary text, its margins and padding are inconsistent with those of the hyperlinks in the list.

**Proposed solution**

Introduce a class `info` and the corresponding CSS in the Contextual module so that informational elements could be added like this:

`<span class="info">Some Informational Text</span>`

And then they would appear with the same padding/margins as the hyperlinks (but not the hover behavior of hyperlinks).

**Additional information**

The need for informational items in contextual links [was discussed here](https://github.com/backdrop/backdrop-issues/issues/2626).

To see an example, paste this code as "Raw HTML" into a node, then hover over the contextual links trigger:

```
<div class="contextual-links-region">
  <div class="contextual-links-wrapper">
    <ul class="contextual-links">
      <li><span class="info">Sample info</span></li>
      <li><a href="#">Child link</a></li>
    </ul>
  </div>
  <div style="font-size: 10px; height: 45px; background-color: #EEE;">Contextual links demo</div>
</div>
```

**Comment by olafgrabienski** [comment]
This will be useful! I'd suggest to also change the appearance of the informational items slightly, so that they don't get confused with links. Below are some screenshots with ideas how to style the info item differently.

**Current PR:**

<img width="707" alt="backdrop-context-links-info-1-pr" src="https://user-images.githubusercontent.com/14188307/135757654-35ad8f32-51d2-40a4-932d-5b5479a9c6be.png">

**Altenative ideas:**

<img width="707" alt="backdrop-context-links-info-2" src="https://user-images.githubusercontent.com/14188307/135757658-016990a4-b201-402e-8a24-69f1e5b7c296.png">
<img width="707" alt="backdrop-context-links-info-3" src="https://user-images.githubusercontent.com/14188307/135757659-d7bd1649-0d67-4b4a-b813-752c4d199c47.png">
<img width="707" alt="backdrop-context-links-info-4" src="https://user-images.githubusercontent.com/14188307/135757660-72f1faee-c82c-4563-a0c0-7a62b67a5e83.png">

**Comment by bugfolder** [comment]
I agree with the separate styling. I was already pondering something like your second example—add gray background only. Seeing the variety  of examples you've provided, I'm now leaning more strongly toward #2. The gray background resembles things that are disabled, as in, "this thing isn't clickable," which is what we want to convey. In #3, the bolding draws extra attention to it, and in #3, the more strongly highlighted background suggests to me the hover rollover.

**Comment by bugfolder** [comment]
PR updated. It looks like this.

![contextual_links_demo](https://user-images.githubusercontent.com/29214084/135773139-531e992e-71bb-4929-a32f-70a881bb2a45.jpg)

**Comment by stpaultim** [comment]
I like the idea of separate styling. My initial reaction was that option #2 is a little too simple and easy to miss. I think I like #3 or #4 better, because they are more noticable. But, I'm not really very confident in my opinion on this and have not thought through the design implications very well. 

I did try the sandbox and just don't feel like the difference between the Description on the links was that noticable. This might be because it's on top of a background that is about the same color. So maybe the example is misleading.

**Comment by indigoxela** [comment]
When writing themes, color definitions in system CSS are inconvenient. Above examples fit Seven, but no other (admin) theme.

I'd like to suggest, that color definitions should *always* happen in themes (Seven, Basis) and not in core module CSS? The views and layout module are already a pain regarding that. Let's not increase the theming work - requiring to write against seven-ish colors all over the place, please.

**Comment by olafgrabienski** [comment]
> I did try the sandbox and just don't feel like the difference between the Description on the links was that noticable. This might be because it's on top of a background that is about the same color. So maybe the example is misleading.

The example is indeed misleading, as it puts a gray background to the content. Usually, you don't find this in *Basis*. The screenshots in https://github.com/backdrop/backdrop-issues/issues/5274#issuecomment-932961505 are realistic.

**Comment by bugfolder** [comment]
All good points, thanks. So it looks like either #3 or #4 are preferred, and the colors should be defined in core themes (Seven, Basis, Bartik) rather than in module file contextual/css/contextual.css.

Any votes on #3 vs #4, or other alternatives to propose? Should the background colors be different in the different themes?

**Comment by olafgrabienski** [comment]
> Any votes on #3 vs #4, or other alternatives to propose? Should the background colors be different in the different themes?

I like No. 2 or 3 for themes with white content background (Basis and Bartik). No. 4 could be a good choice for Seven's light gray background.

**Comment by bugfolder** [comment]
Incidentally, my sample above was No. 2 on Basis. I'd put some color on the background just to identify the contextual links region. Here it is again with a white background

![sample_info](https://user-images.githubusercontent.com/29214084/135891845-45fee803-8fab-4476-be3b-cbd570fa4b69.jpg)

**Comment by stpaultim** [comment]
> I'd like to suggest, that color definitions should always happen in themes (Seven, Basis) and not in core module CSS.

I see the logic here and it makes sense. But, how likely are contrib theme developers to actually style a tiny administrative tool? 

I suppose, at least we should provide the color in the core themes.

**Comment by indigoxela** [comment]
> But, how likely are contrib theme developers to actually style a tiny administrative tool?

Think of admin themes with a "dark mode" or Color module support. Think of frontend themes that are also supposed to work as admin theme.

> ...at least we should provide the color in the core themes.

That's exactly what I'm suggesting: put color definitions into themes. :wink:

**Comment by bugfolder** [comment]
> That's exactly what I'm suggesting: put color definitions into themes.

There's already 3 "background-color:" declarations in core module file contextual.css, so presumably those, too, would have to be moved into themes, as well as the one we're currently considering.

A quick search through the core/modules directory turns up 133 instances of "background-color:" in the CSS (and lots more of just plain "color:"). Being fully consistent about "colors only in themes" sounds like it would be a challenging task with plenty of backward-compatibility issues for people whose theme tree didn't lead back to core themes.

It seems, though, like as long as we're providing lots of default color schemes in core modules, it doesn't make things too bad to add one more as the default. Since No. 2 seems to work for Basis and Bartik (and lots of other sites), how about we provide that in the module, then override with No. 4 in Seven?

Maybe then a task for 2.0 could be to move all colors from core modules into themes.

**Comment by indigoxela** [comment]
@bugfolder I didn't want to suggest, that we cleanup everything now (uh-oh!). Yes, I'm aware that many core modules define their seven-ish or othewise grey-ish colors (already mentioned Views and Layout).

I just wanted to suggest, that we **consider** colors as something theme-specific, not module-specific. Just to not add more themer's pain.

**Comment by bugfolder** [comment]
>  I was thinking for example that we could style these same as table headers for each theme?

A first use case for this is giving the name of the layout as in https://github.com/backdrop/backdrop-issues/issues/2626#issuecomment-929384914:

![135126627-28c5c696-5c45-4a17-b239-9c9ed79bee1f](https://user-images.githubusercontent.com/29214084/136452536-043a087e-494e-42e1-a929-6d1038995107.jpg)

Using the table header element styling would change the case of the layout title to all caps in Seven.
